packages/agreement-process/src/model/domain/toEvent.ts

```diff
@@ -69,19 +74,80 @@ export const toAgreementV1 = (input: Agreement): AgreementV1 => ({
   stamps: toStampsV1(input.stamps),
 });
 
+export const toAgreementStateV2 = (state: AgreementState): AgreementStateV2 =>
+  match(state)
+    .with("Draft", () => AgreementStateV2.DRAFT)
+    .with("Suspended", () => AgreementStateV2.SUSPENDED)
+    .with("Archived", () => AgreementStateV2.ARCHIVED)
+    .with("Pending", () => AgreementStateV2.PENDING)
+    .with("Active", () => AgreementStateV2.ACTIVE)
+    .with("Rejected", () => AgreementStateV2.REJECTED)
+    .with(
+      "MissingCertifiedAttributes",
+      () => AgreementStateV2.MISSING_CERTIFIED_ATTRIBUTES
+    )
+    .exhaustive();
+
+export const toAgreementDocumentV2 = (
+  input: AgreementDocument
+): AgreementDocumentV2 => ({
+  ...input,
+  createdAt: BigInt(input.createdAt.getTime()),
+});
+
+export const toAgreementStampV2 = (
+  input: AgreementStamp
+): AgreementStampV2 => ({
+  ...input,
+  when: BigInt(input.when.getTime()),
+});
+
+export const toAgreementStampsV2 = (
+  input: AgreementStamps
+): AgreementStampsV2 => ({
+  submission: input.submission
+    ? toAgreementStampV2(input.submission)
+    : undefined,
+  activation: input.activation
+    ? toAgreementStampV2(input.activation)
+    : undefined,
+  rejection: input.rejection ? toAgreementStampV2(input.rejection) : undefined,
+  suspensionByProducer: input.suspensionByProducer
+    ? toAgreementStampV2(input.suspensionByProducer)
+    : undefined,
+  upgrade: input.upgrade ? toAgreementStampV2(input.upgrade) : undefined,
+  archiving: input.archiving ? toAgreementStampV2(input.archiving) : undefined,
+  suspensionByConsumer: input.suspensionByConsumer
+    ? toAgreementStampV2(input.suspensionByConsumer)
+    : undefined,
+});
+
+export const toAgreementV2 = (input: Agreement): AgreementV2 => ({
+  ...input,
+  state: toAgreementStateV2(input.state),
+  createdAt: BigInt(input.createdAt.getTime()),
+  updatedAt: input.updatedAt ? BigInt(input.updatedAt.getTime()) : undefined,
```

Comment:
@MalpenZibo if you find it useful, we added some [utils](https://github.com/pagopa/interop-be-monorepo/pull/366/files#diff-1fe0dbc187fdedcc4032c93a9edcd6a86338fc21a4fc861f8aaac759ef16ab1cR1-R11) for date conversion

Reply:
Yes, but I prefer to make this changes in a dedicated PR

Reply:
https://pagopa.atlassian.net/browse/IMN-505